client/js/serverstate.js

```diff
@@ -104,9 +104,13 @@ function receiveDelta(delta) {
     if(delta.s[widgetID] && delta.s[widgetID].parent !== undefined && widgets.has(widgetID))
       $('#topSurface').appendChild(widgets.get(widgetID).domElement);
 
-  for(const widgetID in delta.s)
-    if(delta.s[widgetID] !== null && !widgets.has(widgetID))
+  const widgetWasAdded = false;
```

Comment:
It breaks if another client adds a widget. Not sure why locally added widgets don't go through that code.

Reply:
I have not been able to get that code branch to execute (the branch that sets it to `true`). When widgets are added locally, I think they are already in the `widgets` map before this code gets executed, so `widgetWasAdded` doesn't ever get set to `true`.

Reply:
I think that stuff can just be removed. I added a condition in jeUpdateTree anyway that triggers a complete rebuild if the ID is not found in the tree. I did that because I'm not sure if certain characters in the ID make it fail to find it.

Reply:
Changed from `const` to `let`.

Reply:
Done